### PR TITLE
Fix HTTP status code when output class is set to null

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

Also fix Behat test for when input class is set to `null`.

### CODE REVIEW

File: src/EventListener/SerializeListener.php [hunk 1]
```diff
@@ -61,15 +61,8 @@ public function onKernelView(GetResponseForControllerResultEvent $event): void
...
-        if (
-            (isset($context['output']) && \array_key_exists('class', $context['output']) && null === $context['output']['class'])
-            ||
-            (
-                null === $controllerResult && isset($context['input']) && \array_key_exists('class', $context['input']) &&
-                null === $context['input']['class']
-            )
-        ) {
-            $event->setControllerResult('');
+        if (isset($context['output']) && \array_key_exists('class', $context['output']) && null === $context['output']['class']) {
```
Comment (antograssiot) [inline src/EventListener/SerializeListener.php]: maybe 

```suggestion
        if (\array_key_exists('class', $context['output'] ?? []) && null === $context['output']['class']) {
```

even if it is not directly linked to your change

File: src/Metadata/Resource/Factory/InputOutputResourceMetadataFactory.php [hunk 2]
```diff
@@ -64,6 +64,15 @@ private function getTransformedOperations(array $operations, array $resourceAttr
 
             $operation['input'] = isset($operation['input']) ? $this->transformInputOutput($operation['input']) : $resourceAttributes['input'];
             $operation['output'] = isset($operation['output']) ? $this->transformInputOutput($operation['output']) : $resourceAttributes['output'];
+
+            if (
+                !isset($operation['status'])
+                && isset($operation['output'])
+                && \array_key_exists('class', $operation['output'])
```
Comment (antograssiot) [inline src/Metadata/Resource/Factory/InputOutputResourceMetadataFactory.php]:
```suggestion
                && \array_key_exists('class', $operation['output'] ?? [])
```

and drop the isset before ?

Comment (soyuka) [reply src/Metadata/Resource/Factory/InputOutputResourceMetadataFactory.php]: I'd be to keep isset, it's readable and doesn't harm no?

File: src/EventListener/SerializeListener.php (hunk 1 above)
Comment (soyuka) [reply src/EventListener/SerializeListener.php]: I'd be to keep isset, it's readable and doesn't harm no?

File: src/Metadata/Resource/Factory/InputOutputResourceMetadataFactory.php (hunk 2 above)
Comment (antograssiot) [reply src/Metadata/Resource/Factory/InputOutputResourceMetadataFactory.php]: ok

**Comment by soyuka** [comment]
Thanks @teohhanhui !
